### PR TITLE
Added signing support to FObjects

### PR DESCRIPTION
**Pull request by kirkeaton**
1. Added Signer interface
2. Added Signable interface
3. Made FObject and PropertyInfo implement Signable and Signer respectively
4. Added Signing Benchmark test

### CODE REVIEW

File: src/foam/core/AbstractFObject.java [hunk 1]
```diff
@@ -144,4 +144,29 @@ public boolean hasDefaultValue(String prop) {
...
     }
   }
+
+  public byte[] sign(PrivateKey key) {
+    return this.sign("SHA256withRSA", key);
+  }
+
+  public byte[] sign(String algorithm, PrivateKey key) {
+    try {
+      Signature signature = Signature.getInstance(algorithm);
```
Comment (kgrgreer) [inline src/foam/core/AbstractFObject.java:154]: How expensive are these two lines? Can we reuse these by just passing in the Signature?

Comment (kirkeaton) [reply src/foam/core/AbstractFObject.java:154]: It didn't make an effect in terms of performance.

File: src/foam/nanos/bench/SigningBenchmark.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.nanos.bench;
+
+import foam.core.X;
+import foam.dao.DAO;
+import foam.dao.ListSink;
+import foam.dao.Sink;
+import net.nanopay.tx.model.Transaction;
```
Comment (adamvy) [inline src/foam/nanos/bench/SigningBenchmark.java:13]: This adds nanopay dependencies to foam2.  Can we add this class to the NANOPAY repo instead?  Standalone nanos is broken as a result.

Comment (kirkeaton) [reply src/foam/nanos/bench/SigningBenchmark.java:13]: Yeah, my mistake.

**Pull request merged**